File: rabbitmq-dotnet-client/Send.cs


using System.Text;
using RabbitMQ.Client;



ConnectionFactory factory = new ConnectionFactory();
// "guest"/"guest" by default, limited to localhost connections
factory.UserName = "admin";
factory.Password = "password";
factory.VirtualHost = "/";
factory.Port=8888;
factory.HostName = "localhost";

// IConnection conn = factory.CreateConnection();



// var factory = new ConnectionFactory { HostName = "localhost" };
using var connection = factory.CreateConnection();
using var channel = connection.CreateModel();

channel.QueueDeclare(queue: "hello",
                     durable: false,
                     exclusive: false,
                     autoDelete: false,
                     arguments: null);

var message = "Hello World!";
var body = Encoding.UTF8.GetBytes(message);

channel.BasicPublish(exchange: string.Empty,
                     routingKey: "hello",
                     basicProperties: null,
                     body: body);
Console.WriteLine($" [x] Sent {message}");

Console.WriteLine(" Press [enter] to exit.");
Console.ReadLine();

File: rabbitmq-pika-client/send.py
import pika
import time
from datetime import datetime


username = 'admin'
password = 'password'
rabbitmq_password  = pika.PlainCredentials(username,password)



# conn =  pika.BlockingConnection(pika.ConnectionParameters('localhost'))
# conn =  pika.BlockingConnection(pika.ConnectionParameters('localhost',credentials=rabbitmq_password))
conn =  pika.BlockingConnection(pika.ConnectionParameters('localhost','5672','/',credentials=rabbitmq_password))


chan = conn.channel()

# chan.queue_declare(queue='qq1',durable=True)
chan.queue_declare(queue='qq1',durable=True,arguments={"x-queue-type": "quorum"})






for i in range(10000):
    now = datetime.now()

    # dt_string = now.strftime("%d/%m/%Y %H:%M:%S")
    dt_string = str(i)


    chan.basic_publish(exchange='',
                       routing_key='qq1',
                       body='hello world 1111' + dt_string,mandatory=True)
    chan.confirm_delivery()



    print("message sent!")


conn.close()

